File: go/hackerrank/gradingStudents.go
// https://www.hackerrank.com/challenges/grading/problem

func gradingStudents(grades []int32) []int32 {
    result := []int32{}
    for _, g := range(grades){
        if g < 38 {
            result = append(result, g)
            continue
        }
        m := g % 10
        if m < 5 {
            if 5 - m < 3 {
                g += 5 - m
            }
        } else {
            if 10 - m < 3 {
                g += 10 - m
            }
        }
        result = append(result, g)
    }
    return result
}

File: python/hackerrank/grading_students.py
# https://www.hackerrank.com/challenges/grading/problem

def gradingStudents(grades):
    result = []
    for grade in grades:
        if grade < 38:
            result.append(grade)
            continue

        mod = grade % 10
        if mod < 5:
            grade = grade + 5 - mod if 5 - mod < 3 else grade
        else:
            grade = grade + 10 - mod if 10 - mod < 3 else grade

        result.append(grade)
    return result
